<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="41" passed="41" failed="0" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2023-05-07T13:24:21 IST" name="Default suite" finished-at="2023-05-07T13:24:21 IST" duration-ms="241">
    <groups>
    </groups>
    <test started-at="2023-05-07T13:24:21 IST" name="Default test" finished-at="2023-05-07T13:24:21 IST" duration-ms="241">
      <class name="com.nityaobject.roombooking.NewTestRoom">
        <test-method signature="testingBookingPositive(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:1, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingPositive" data-provider="GetSampleDataForBookingPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="39" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=1, roomRate=150.0, numberOfguests=0]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingPositive -->
        <test-method signature="testingBookingPositive(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:1, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingPositive" data-provider="GetSampleDataForBookingPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="3" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=2, roomRate=150.0, numberOfguests=1]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=2, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingPositive -->
        <test-method signature="testingBookingPositive(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:1, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingPositive" data-provider="GetSampleDataForBookingPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[6/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[10/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=2, roomRate=150.0, numberOfguests=1]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Vijay, customer_id=2}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=2, room_Id=2, startDate=Thu Jan 06 00:00:00 IST 2022, endDate=Mon Jan 10 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingPositive -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=1, roomRate=150.0, numberOfguests=0]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[4/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=1, roomRate=150.0, numberOfguests=0]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[3/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[99/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=1, roomRate=150.0, numberOfguests=0]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=2, roomRate=150.0, numberOfguests=1]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="3" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=1, roomRate=150.0, numberOfguests=0]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingBookingNegative(java.lang.String,java.lang.String,com.nityaobject.roombooking.Room,com.nityaobject.roombooking.Customer,com.nityaobject.roombooking.BookingDetails)[pri:2, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingBookingNegative" data-provider="GetSampleDataForBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022 ]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[Room [roomId=2, roomRate=150.0, numberOfguests=1]]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
            <param index="4">
              <value>
                <![CDATA[BookingDetails{customer_Id=1, room_Id=2, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.AlreadyBookedDates">
            <message>
              <![CDATA[Already Booked]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.AlreadyBookedDates: Already Booked
at com.nityaobject.roombooking.RoomBooking.bookingRoom(RoomBooking.java:92)
at com.nityaobject.roombooking.NewTestRoom.testingBookingNegative(NewTestRoom.java:100)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.AlreadyBookedDates -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingBookingNegative -->
        <test-method signature="testingOfFindBookingByRoomIdPositive(int,java.util.List)[pri:3, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdPositive" data-provider="GetSampleDataToFindBookingsRoomIdPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="7" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdPositive -->
        <test-method signature="testingOfFindBookingByRoomIdPositive(int,java.util.List)[pri:3, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdPositive" data-provider="GetSampleDataToFindBookingsRoomIdPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022, Thu Jan 06 00:00:00 IST 2022, Fri Jan 07 00:00:00 IST 2022, Sat Jan 08 00:00:00 IST 2022, Sun Jan 09 00:00:00 IST 2022, Mon Jan 10 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdPositive -->
        <test-method signature="testingOfFindBookingByRoomIdNegative(int,java.util.List)[pri:4, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdNegative" data-provider="GetSampleDataToFindBookingsRoomIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[3]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.findBookingByRoomId(RoomBooking.java:153)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByRoomIdNegative(NewTestRoom.java:211)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdNegative -->
        <test-method signature="testingOfFindBookingByRoomIdNegative(int,java.util.List)[pri:4, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdNegative" data-provider="GetSampleDataToFindBookingsRoomIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="0" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[4]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022, Thu Jan 06 00:00:00 IST 2022, Fri Jan 07 00:00:00 IST 2022, Sat Jan 08 00:00:00 IST 2022, Sun Jan 09 00:00:00 IST 2022, Mon Jan 10 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.findBookingByRoomId(RoomBooking.java:153)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByRoomIdNegative(NewTestRoom.java:211)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdNegative -->
        <test-method signature="testingOfFindBookingByRoomIdNegative(int,java.util.List)[pri:4, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdNegative" data-provider="GetSampleDataToFindBookingsRoomIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[3]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022, Thu Jan 06 00:00:00 IST 2022, Fri Jan 07 00:00:00 IST 2022, Sat Jan 08 00:00:00 IST 2022, Sun Jan 09 00:00:00 IST 2022, Mon Jan 10 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.findBookingByRoomId(RoomBooking.java:153)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByRoomIdNegative(NewTestRoom.java:211)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdNegative -->
        <test-method signature="testingOfFindBookingByRoomIdNegative(int,java.util.List)[pri:4, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByRoomIdNegative" data-provider="GetSampleDataToFindBookingsRoomIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[4]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[Sat Jan 01 00:00:00 IST 2022, Sun Jan 02 00:00:00 IST 2022, Mon Jan 03 00:00:00 IST 2022, Tue Jan 04 00:00:00 IST 2022, Wed Jan 05 00:00:00 IST 2022]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.findBookingByRoomId(RoomBooking.java:153)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByRoomIdNegative(NewTestRoom.java:211)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByRoomIdNegative -->
        <test-method signature="testingOfFindBookingByCustomerIdPositive(int,java.util.List)[pri:5, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByCustomerIdPositive" data-provider="GetSampleDataToFindBookingsCustomerIdPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}, BookingDetails{customer_Id=1, room_Id=2, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByCustomerIdPositive -->
        <test-method signature="testingOfFindBookingByCustomerIdPositive(int,java.util.List)[pri:5, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByCustomerIdPositive" data-provider="GetSampleDataToFindBookingsCustomerIdPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[BookingDetails{customer_Id=2, room_Id=2, startDate=Thu Jan 06 00:00:00 IST 2022, endDate=Mon Jan 10 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByCustomerIdPositive -->
        <test-method signature="testingOfFindBookingByCustomerIdNegative(int,java.util.List)[pri:6, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByCustomerIdNegative" data-provider="GetSampleDataToFindBookingsCustomerIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[3]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[BookingDetails{customer_Id=2, room_Id=2, startDate=Thu Jan 06 00:00:00 IST 2022, endDate=Mon Jan 10 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.findBookingByCustomerId(RoomBooking.java:141)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByCustomerIdNegative(NewTestRoom.java:266)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByCustomerIdNegative -->
        <test-method signature="testingOfFindBookingByCustomerIdNegative(int,java.util.List)[pri:6, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingOfFindBookingByCustomerIdNegative" data-provider="GetSampleDataToFindBookingsCustomerIdNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="0" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[4]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[[BookingDetails{customer_Id=1, room_Id=1, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=1045.0, extraGuestsOfCustomer=0}, BookingDetails{customer_Id=1, room_Id=2, startDate=Sat Jan 01 00:00:00 IST 2022, endDate=Wed Jan 05 00:00:00 IST 2022, status=true, totalAmountOfCustomer=3520.0, extraGuestsOfCustomer=1}]]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.findBookingByCustomerId(RoomBooking.java:141)
at com.nityaobject.roombooking.NewTestRoom.testingOfFindBookingByCustomerIdNegative(NewTestRoom.java:266)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingOfFindBookingByCustomerIdNegative -->
        <test-method signature="testingCalculateAmountBookingPositive(java.lang.String,java.lang.String,int,double)[pri:7, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingPositive" data-provider="GetSampleDataToCalculateBookingsPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="17" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[1045]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingPositive -->
        <test-method signature="testingCalculateAmountBookingPositive(java.lang.String,java.lang.String,int,double)[pri:7, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingPositive" data-provider="GetSampleDataToCalculateBookingsPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[6/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[10/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[3520]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingPositive -->
        <test-method signature="testingCalculateAmountBookingPositive(java.lang.String,java.lang.String,int,double)[pri:7, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingPositive" data-provider="GetSampleDataToCalculateBookingsPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[3520]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingPositive -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[6/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[10/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[4]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[3520]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:265)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[6/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[10/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[1045]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:293)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[11/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[155/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[3520]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:293)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[11/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[15/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[3520]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:293)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="0" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[4]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[1045]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:265)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testingCalculateAmountBookingNegative(java.lang.String,java.lang.String,int,double)[pri:8, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testingCalculateAmountBookingNegative" data-provider="GetSampleDataToCalculateBookingsNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[6/1/2022]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[10/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[1045]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.calculateThetotalbillOfARoomFromParticularDate(RoomBooking.java:293)
at com.nityaobject.roombooking.NewTestRoom.testingCalculateAmountBookingNegative(NewTestRoom.java:311)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testingCalculateAmountBookingNegative -->
        <test-method signature="testCancelBookingPositive(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:9, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingPositive" data-provider="GetSampleDataForCancellingBookingPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="8" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingPositive -->
        <test-method signature="testCancelBookingPositive(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:9, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingPositive" data-provider="GetSampleDataForCancellingBookingPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingPositive -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[3]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidRoom">
            <message>
              <![CDATA[ Room not present ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidRoom:  Room not present 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:204)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidRoom -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Vijay, customer_id=2}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:194)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Vijay, customer_id=2}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:194)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="2" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[8/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:199)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[4/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Vijay, customer_id=2}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:194)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[8/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[12/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Vijay, customer_id=2}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidDateException">
            <message>
              <![CDATA[ Invalid Date ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidDateException:  Invalid Date 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:194)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidDateException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[3/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:199)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="0" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[2]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:199)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelBookingNegative(int,java.lang.String,java.lang.String,com.nityaobject.roombooking.Customer)[pri:10, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelBookingNegative" data-provider="GetSampleDataForCancellingBookingNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[1]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[1/1/2022]]>
              </value>
            </param>
            <param index="2">
              <value>
                <![CDATA[5/1/2022]]>
              </value>
            </param>
            <param index="3">
              <value>
                <![CDATA[Customer{customer_name=Ajay, customer_id=1}]]>
              </value>
            </param>
          </params>
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:199)
at com.nityaobject.roombooking.NewTestRoom.testCancelBookingNegative(NewTestRoom.java:342)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelBookingNegative -->
        <test-method signature="testCancelRoomforARangeOfdatesUsingObjectPositive()[pri:11, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelRoomforARangeOfdatesUsingObjectPositive" finished-at="2023-05-07T13:24:21 IST" duration-ms="3" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelRoomforARangeOfdatesUsingObjectPositive -->
        <test-method signature="testCancelRoomforARangeOfdatesUsingObjectNegative()[pri:12, instance:com.nityaobject.roombooking.NewTestRoom@be64738]" started-at="2023-05-07T13:24:21 IST" name="testCancelRoomforARangeOfdatesUsingObjectNegative" finished-at="2023-05-07T13:24:21 IST" duration-ms="0" status="PASS">
          <exception class="com.nityaobject.roombooking.InvalidCustomer">
            <message>
              <![CDATA[ Invalid Customer ]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.nityaobject.roombooking.InvalidCustomer:  Invalid Customer 
at com.nityaobject.roombooking.RoomBooking.cancelRoomforASpecificRangeOfdates(RoomBooking.java:199)
at com.nityaobject.roombooking.RoomBooking.cancelRoomforARangeOfdatesUsingObject(RoomBooking.java:247)
at com.nityaobject.roombooking.NewTestRoom.testCancelRoomforARangeOfdatesUsingObjectNegative(NewTestRoom.java:376)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:577)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- com.nityaobject.roombooking.InvalidCustomer -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCancelRoomforARangeOfdatesUsingObjectNegative -->
      </class> <!-- com.nityaobject.roombooking.NewTestRoom -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
